spark/src/main/scala/io/substrait/spark/logical/ToLogicalPlan.scala

```diff
@@ -409,23 +433,98 @@ class ToLogicalPlan(spark: SparkSession = SparkSession.builder().getOrCreate())
       case format =>
         throw new UnsupportedOperationException(s"File format not currently supported: $format")
     }
-    new LogicalRelation(
-      relation = HadoopFsRelation(
-        location = new InMemoryFileIndex(
-          spark,
-          localFiles.getItems.asScala.map(i => new Path(i.getPath.get())),
-          Map(),
-          Some(schema)),
-        partitionSchema = new StructType(),
-        dataSchema = schema,
-        bucketSpec = None,
-        fileFormat = format,
-        options = options
-      )(spark),
-      output = output,
-      catalogTable = None,
-      isStreaming = false
+  }
+
+  override def visit(write: NamedWrite, context: EmptyVisitationContext): LogicalPlan = {
+    val child = write.getInput.accept(this, context)
+    val names = write.getNames
+    val (table, database, catalog) = names.size match {
+      case 1 => (names.get(0), None, None)
+      case 2 => (names.get(1), Some(names.get(0)), None)
+      case 3 => (names.get(2), Some(names.get(1)), Some(names.get(0)))
+      case _ =>
+        throw new UnsupportedOperationException(
+          s"NamedWrite requires up to three names ([[catalog,] database,] table): $names")
+    }
```

Comment:
I'm not sure it is better but I _think_ you could avoid the index access and give the elements suitable names with:

```scala
    val (table, database, catalog) = names.asScala.toList match {
      case List(table) => (table, None, None)
      case List(database, table) => (table, Some(database), None)
      case List(catalog, database, table) => (table, Some(database), Some(catalog))
      case _ =>
        throw new UnsupportedOperationException(
          s"NamedWrite requires up to three names ([[catalog,] database,] table): $names")
    }
```